astropy/cosmology/tests/test_core.py

```diff
@@ -400,13 +400,32 @@ class TestFlatSomeCosmology(FlatCosmologyMixinTest, TestSomeCosmology):
             ...
     """
 
+    def test_nonflat_class_(self, cosmo_cls, cosmo):
+        """Test :attr:`astropy.cosmology.core.FlatCosmologyMixin.nonflat_cls`.
+        """
+        # Test it's a method on the class
+        assert issubclass(cosmo_cls, cosmo_cls._nonflat_cls_)
+
+        # It also works from the instance. # TODO! as a "metaclassmethod"
+        assert issubclass(cosmo_cls, cosmo._nonflat_cls_)
+
+        # Maybe not the most robust test, but so far all Flat classes have the
+        # name of their parent class.
+        assert cosmo._nonflat_cls_.__name__ in cosmo_cls.__name__
+
     def test_is_flat(self, cosmo_cls, cosmo):
         """Test property ``is_flat``."""
         super().test_is_flat(cosmo_cls, cosmo)
 
         # it's always True
         assert cosmo.is_flat is True
 
+    def test_equivalent_nonflat(self, cosmo):
+        """Test :attr:`astropy.cosmology.core.FlatCosmologyMixin.equivalent_nonflat`.
+        """
+        assert cosmo.equivalent_nonflat.is_equivalent(cosmo)
+        assert cosmo.is_equivalent(cosmo.equivalent_nonflat)
```

Comment:
``is_equivalent`` is commutative for Cosmology instances because it mimics the Python `__eq__` behavior by routing to the other instance's equivalence method when `NotImplemented` is returned.
So while it is tested elsewhere, it's not tested for each flat cosmology in this way.